### PR TITLE
Converter

### PR DESCRIPTION
**Pull request by nicktate**
Adds `create-from-file` support to the `svc` command. Welcome feedback on command format, this was the initially quick and dirty version to get it testing. 

* Possibly make file inputs top level argument for command rather than flag?

## THIS PR IS NOT MERGEABLE. https://github.com/containership/containership.schema-converter/pull/2 must be merged and then the repository needs cut and updated in this package json in order for this PR to be ready to merge.

> In order to test locally, you can checkout this PR branch and then manually `npm link ../path/to/containership.schema` locally assuming you have already cloned that repo.

@normanjoyner @ashleyschuett @mikestoltz @phildougherty

### CODE REVIEW

File: commands/svc.js [hunk 1]
```diff
@@ -191,6 +193,54 @@ module.exports.commands.push({
...
+        }
+    },
+    callback: (argv) => {
+        const config = Converter.getConverter(argv.file, {
+            configuration_service: argv.type.toLowerCase(),
+            configuration_type: 'file'
+        });
+
+        const result = config.convertTo(Converter.services.CONTAINERSHIP);
+        result.errors.web.push('what is happening now');
```
Comment (normanjoyner) [inline commands/svc.js]: remove debugging

File: commands/svc.js [hunk 2]
```diff
@@ -191,6 +193,54 @@ module.exports.commands.push({
...
+            alias: 'f'
+        },
+        type: {
+            required: true,
+            description: 'Type of configuration. (docker, containership, kubernetes)',
+            alias: 't'
+        }
+    },
+    callback: (argv) => {
+        const config = Converter.getConverter(argv.file, {
```
Comment (normanjoyner) [inline commands/svc.js]: we should catch error or validate `argv.type`

File: commands/svc.js [hunk 3]
```diff
@@ -191,6 +193,54 @@ module.exports.commands.push({
...
+
+        // if there were any parsing errors, warn about them
+        _.forEach(result.errors, (errorList, app) => {
+            if(errorList.length > 0) {
+                logger.warn(`Service ${app} encountered the following errors during parsing:`);
+                logger.warn(`\t${errorList.join('\n\t')}`);
+            }
+        });
+
+        return request.post('applications', {}, result.configuration, (err, response) => {
```
Comment (normanjoyner) [inline commands/svc.js]: should we post a potentially half accepted configuration, or just bomb out completely?

File: commands/svc.js (hunk 2 above)
Comment (nicktate) [reply commands/svc.js]: happens above in options configuration

**Comment by ashleyschuett** [comment]
You currently have both `package-lock.json` file and a `yarn.lock` file which is confusing.

**Comment by ashleyschuett** [comment]
https://containership.github.io/containership.cli/docs/quick-start-guide/

after `csctl --help` the list of commands isn't alphabetical

**Comment by ashleyschuett** [comment]
The paths in the docs have you name.. `/Users/nick/.containership/plugins`

**Comment by ashleyschuett** [comment]
`csdemo svc create-from-file --f ../containership.schema-converter/examples/dockerExampleV2.yml -t docker`
```/Users/ashley/ContainerShip/containership.cli/lib/configuration.js:21
    return conf.remotes[active];
                       ^

TypeError: Cannot read property 'undefined' of undefined
    at Object.module.exports.getActiveCluster (/Users/ashley/ContainerShip/containership.cli/lib/configuration.js:21:24)
    at getAndCheckActiveCluster (/Users/ashley/ContainerShip/containership.cli/lib/request.js:48:41)
    at requestOptions (/Users/ashley/ContainerShip/containership.cli/lib/request.js:33:27)
    at Object.module.exports.post (/Users/ashley/ContainerShip/containership.cli/lib/request.js:14:19)
    at async.waterfall (/Users/ashley/ContainerShip/containership.cli/commands/svc.js:264:32)
    at nextTask (/Users/ashley/ContainerShip/containership.cli/node_modules/async/dist/async.js:5273:14)
    at next (/Users/ashley/ContainerShip/containership.cli/node_modules/async/dist/async.js:5280:9)
    at /Users/ashley/ContainerShip/containership.cli/node_modules/async/dist/async.js:906:16
    at Immediate.setImmediate (/Users/ashley/ContainerShip/containership.cli/commands/svc.js:245:32)
    at runCallback (timers.js:637:20)```

**Comment by ashleyschuett** [comment]
Got this message`Active cluster must contain an `version`!`

https://github.com/containership/containership.cli/blob/63affaeb73601fac504f6f3030f4772a19a8126c/lib/request.js#L56-L61

should be changed to `Active cluster must contain a `version`!`

**Comment by ashleyschuett** [comment]
`csctl remote add <remote_name> <remote_url>`

I think we should maybe give an example of a remote_url, or what hitting that url should return to be a valid cs cluster that will work with the cli?

**Pull request merged**